omod/src/main/webapp/resources/dhisconnector-runreports.js

```diff
@@ -41,72 +42,103 @@ function populateReportsDropdown() {
 }
 
 function onMappingSelect() {
+    selectedMapping = mappings[jQuery('#mappingSelect').val()];
     // clear the date
     document.getElementById('custom-range-option').checked = false;
     jQuery('#un-recognized-period').html('');
     toggleCustomRangeCheckbox(false);
     jQuery('#periodSelector').val("");
     jQuery('#periodSelector').monthpicker('destroy');
     jQuery('#periodSelector').datepicker('destroy');
-
     jQuery('#ui-datepicker-div > table > tbody > tr').die('mousemove');
     jQuery('.ui-datepicker-calendar tr').die('mouseleave');
 
     weekStartDate = null;
     weekEndDate = null;
 
-    // get periodType
-    var selectedPeriodType = mappings.filter(function (v) {
-        return v.created == jQuery('#mappingSelect').val();
-    })[0].periodType;
-
-    if (selectedPeriodType == undefined)
-        return;
+    switch(selectedMapping.periodType){
+        case undefined:
+            return ;
+        case 'Daily':
+            jQuery('#dailyperiodSelector').datepicker({
+                dateFormat: 'yymmdd'
+            });
+            document.getElementById("monthly").style.display = "show";
```

Comment:
Can't we show only the div that needs to be shown and stop setting display none for other types everytime